geopandas/io/arrow.py

```diff
@@ -171,10 +176,10 @@ def _create_metadata(df, schema_version=None, write_covering_bbox=False):
         if write_covering_bbox:
             column_metadata[col]["covering"] = {
                 "bbox": {
-                    "xmin": ["bbox", "xmin"],
-                    "ymin": ["bbox", "ymin"],
-                    "xmax": ["bbox", "xmax"],
-                    "ymax": ["bbox", "ymax"],
+                    "xmin": [bbox_encoding_fieldname, "xmin"],
+                    "ymin": [bbox_encoding_fieldname, "ymin"],
+                    "xmax": [bbox_encoding_fieldname, "xmax"],
+                    "ymax": [bbox_encoding_fieldname, "ymax"],
```

Comment:
I know you use this for testing, but I think my preference would be to make the test code a bit longer, while keeping it here simpler (only what is needed for the actual code)

Or at least give it a try to see if it would not be too difficult to do this in the tests. I assume you could do something similar as now in the tests, but after creating the table, rename the bbox column, get the metadata and edit this, and replace the metadata.

Something roughly like:
```python
    table = _geopandas_to_arrow(
        df,
        schema_version="1.1.0",
        write_covering_bbox=True,
    )
    table = table.rename_columns([...])  # this needs to be hardcoded list of names, so maybe easiest to use dummy dataframe instead of naturalearth
    metadata = table.schema.metadata
    geo_metadata =  json.loads(metadata[b"geo"])
    # edit metadata ...
    metadata[b"geo"] = ...
    table = table.replace_schema_metadata(metadata)
    pq.write_table(table, filename)
```

Reply:
Thanks for that steer, that direction is very helpful to get around the hurdle. I have updated the test accordingly and reverted changes to adding the kwarg